### PR TITLE
Added support for updating the release date references in CKEditor 5

### LINKED ISSUE #13972: Release process: Update CKEditor 5 references
## 📝 Provide a description of the new feature

**🎯 The goal is to update data stored in the source code or the documentation files.**

- Examples:
    - CKEditor 5 version is stored in the `packages/ckeditor5-utils/src/version.ts` ([source](https://github.com/ckeditor/ckeditor5/blob/f4d2045c9b60d9c5a7055ca993be53e1e8a1d898/packages/ckeditor5-utils/src/version.ts#L14))
    - Also, we would like to update the `README.md` file containing a CDN link. ([source](https://github.com/ckeditor/ckeditor5#ckeditor-5-predefined-builds))
    - We will have a release date stored in CKEditor 5 code base soon.
- We have a script that updates the CKEditor 5 version in different files. ([source](https://github.com/ckeditor/ckeditor5/blob/f4d2045c9b60d9c5a7055ca993be53e1e8a1d898/scripts/release/update-version-references.js))

---

If you'd like to see this feature implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal: Added support for updating the release date references in CKEditor 5. Closes #13972.

---

### Additional information

Notes:
- Removed the ability to make a commit. In the new release process a commit will be done at the end.
- Now, this script exports a function to make it possible to use it in the new release process.

### CODE REVIEW

**Review by pomek — COMMENTED** [review]
Would you like to integrate the spinner/counter into this script?

**Comment by pomek** [comment]
One more point – perhaps we can integrate this script into the release process? I mean, execute the function in `scripts/release/prepare-packages.js`.

**Comment by psmyrek** [comment]
Spinner integration seems unnecessary, because the script will execute very quickly and no one will even notice the spinner. The spinner would only blink for a short while.

But I integrated the script with the release process.

File: scripts/release/update-version-references.js [hunk 1]
```diff
@@ -12,67 +12,74 @@
...
+		// 	file: ...
+		// 	pattern: ...
+		// 	value: releaseDate
+		// }
+	];
+
+	console.log( chalk.blue( `Updating CKEditor 5 version (${ version }) and release date (${ releaseDate }) references.\n` ) );
+
+	for ( const { file, pattern, value } of filesToUpdate ) {
+		const absolutePath = path.join( ROOT_DIRECTORY, file ).split( path.sep ).join( path.posix.sep );
```
Comment (przemyslaw-zan) [inline scripts/release/update-version-references.js:54]: As discussed recently, we should drop manual path normalization in favor of `upath` package.

**Review by przemyslaw-zan — CHANGES_REQUESTED** [review]
LGTM, with one minor request.

File: scripts/release/update-version-references.js [hunk 2]
```diff
@@ -12,67 +12,74 @@
...
+			file: 'README.md',
+			pattern: /(?<=cdn\.ckeditor\.com\/ckeditor5\/)\d+\.\d+\.\d+(?=\/)/,
+			value: version
+		},
+		{
+			file: path.join( 'packages', 'ckeditor5-utils', 'src', 'version.ts' ),
+			pattern: /(?<=const version = ')\d+\.\d+\.\d+(?=';)/,
+			value: version
+		}
+		// TODO: Add file containing the release date.
```
Comment (martnpaneq) [inline scripts/release/update-version-references.js:43]: Could we please not merge `// TODO` comments? It's likely that we'll forget it anyway. If it has to be done as part of some task, please add it to ZenHub.

File: scripts/release/update-version-references.js [hunk 3]
```diff
@@ -12,67 +12,74 @@
...
-	exec( 'git commit -m "Internal: Updated CKEditor 5 version references."' );
-
-	console.log( chalk.green( `\nUpdated CKEditor 5 version references to ${ cke5version }.` ) );
-} else {
-	console.log( chalk.green( `\nNothing to commit. CKEditor 5 version references are up-to-date (${ cke5version }).` ) );
-}
-
-function exec( command ) {
-	console.log( chalk.gray( `$ ${ command }` ) );
+// TODO: Not sure if the date format is OK.
```
Comment (martnpaneq) [inline scripts/release/update-version-references.js:77]: Please remove `// TODO` before merging :)

**Review by pomek — APPROVED** [review]
LGTM.

**Review by martnpaneq — APPROVED** [review]

**Comment by pomek** [comment]
@przemyslaw-zan, both arguments are marked as required. It's an internal script… we do not verify input as we know who consumes it.

image

**Pull request merged**